spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -71,12 +75,17 @@ public ChannelBindingService(ChannelBindingServiceProperties channelBindingServi
 		List<Binding<MessageChannel>> bindings = new ArrayList<>();
 		Binder<MessageChannel, ConsumerProperties, ?> binder =
 				(Binder<MessageChannel, ConsumerProperties, ?>) getBinderForChannel(inputChannelName);
-		Class<? extends ConsumerProperties> propertiesClass = resolveConsumerPropertiesType(binder);
 		ConsumerProperties consumerProperties =
-				this.channelBindingServiceProperties.getConsumerProperties(inputChannelName, propertiesClass);
+				this.channelBindingServiceProperties.getConsumerProperties(inputChannelName);
+		if (binder instanceof ExtendedPropertiesBinder) {
+			ExtendedPropertiesBinder extendedPropertiesBinder = (ExtendedPropertiesBinder) binder;
+			Object extension = extendedPropertiesBinder.getExtendedConsumerProperties(inputChannelName);
+			ExtendedConsumerProperties extendedConsumerProperties = new ExtendedConsumerProperties(extension);
+			BeanUtils.copyProperties(consumerProperties, extendedConsumerProperties);
+			consumerProperties = extendedConsumerProperties;
```

Comment:
Updated the returned object makes me a little queasy (although I guess it's unique for this consumer) - that said, unless I am missing something, I don't see the registry `setConsumer()` or `setProducer()` methods ever being called so it seems we always get a new instance.